internal/oci/runtime_oci.go

```diff
@@ -898,73 +899,82 @@ func (r *runtimeOCI) AttachContainer(c *Container, inputStream io.Reader, output
 	return nil
 }
 
-// PortForwardContainer forwards the specified port provides statistics of a container.
-func (r *runtimeOCI) PortForwardContainer(c *Container, port int32, stream io.ReadWriter) error {
-	emptyStreamOnError := true
-	defer func() {
-		if emptyStreamOnError {
-			go func() {
-				if _, copyError := pools.Copy(ioutil.Discard, stream); copyError != nil {
-					logrus.Errorf("error closing port forward stream after other error: %v", copyError)
-				}
-			}()
+// PortForwardContainer forwards the specified port into the provided container.
+func (r *runtimeOCI) PortForwardContainer(ctx context.Context, c *Container, netNsPath string, port int32, stream io.ReadWriteCloser) error {
+	log.Infof(ctx,
+		"Starting port forward for %s in network namespace %s", c.ID(), netNsPath,
+	)
+
+	// Adapted reference implementation:
+	// https://github.com/containerd/cri/blob/8c366d/pkg/server/sandbox_portforward_unix.go#L65-L120
+	if err := ns.WithNetNSPath(netNsPath, func(_ ns.NetNS) error {
+		defer stream.Close()
+
+		conn, err := net.Dial("tcp", net.JoinHostPort("127.0.0.1", strconv.Itoa(int(port))))
```

Comment:
sorry for nitpicking, but if you have the chance of changing the PR again I prefer to carry on with the comment and implementation I used in containerd , if you don't have the whole context of the discussion this is easy to be changed by mistake
```suggestion
		// TODO: hardcoded to tcp4 because localhost resolves to ::1 by default if the system has IPv6 enabled.
		// Theoretically happy eyeballs will try IPv6 first and fallback to IPv4
		// but resolving localhost doesn't seem to return and IPv4 address always, thus failing the connection.
		conn, err := net.Dial("tcp4", fmt.Sprintf("localhost:%d", port))```

https://github.com/containerd/cri/pull/1470/files#diff-a60e53338184ff4aee2ff77caed7cabaR67-R70

Reply:
Sure, added the comment as suggested.

Reply:
and the code	`conn, err := net.Dial("tcp4", fmt.Sprintf("localhost:%d", port))` 
:upside_down_face:

Reply:
I was confused, changed that as well.